models/Discord/Commands/DiscordCommandWhoIs.js

```diff
@@ -29,32 +31,54 @@ class DiscordCommandWhoIs extends DiscordCommand {
         return;
       }
       var userID = auser
-      if(!userID) {
-        message.reply("Unknown Error while getting Discord ID!")
-        return;
+      var DB_response;
+      
+      if(userID) {
+        DB_response = getByDiscordID(connection, userID)
+      } else {
+        DB_response = getByCkey(connection, provided_ckey, message)
       }
-
-      connection.query("SELECT * FROM `erro_player` WHERE `discord_id` = ?", [userID.id], (error, results, fields) => {
-        if (error) {
-          message.reply("Error running select query, try again later.");
-          return;
-        }
-        if (results.length == 0) {
-          message.reply("No linked BYOND account found for this user.");
-          return;
-        }
-        if (results.length > 1) {
-          message.reply("More than one BYOND account linked to this ID. This shouldn't happen!");
-          return;
-        }
-
-        var ckey = results[0].ckey
-        message.reply(userID + " belongs to the ckey '" + ckey +"'")
-      })
+        
+      message.reply(DB_response)
       connection.release();
     });
   }
 
 }
 
+function getByDiscordID(connection, user) {
+  connection.query("SELECT * FROM `erro_player` WHERE `discord_id` = ?", [user.id], (error, results, fields) => {
+      if (error) {
+        return "Error running select query, try again later.";
+      }
+      if (results.length == 0) {
+        return "No linked BYOND account found for this user.";
+      }
+      if (results.length > 1) {
+        return "More than one BYOND account linked to this ID. This shouldn't happen!";
+      }
+
+      var ckey = results[0].ckey
+      return (user + " belongs to the ckey '" + ckey +"'")
```

Comment:
This wont actually work as the query results are returned asynchronously so you can't return at this point.

Instead of returning a string here you'll have to reply to the discord message with message.reply()